core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/SyncFrequency.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.stitch.core.services.mongodb.remote.sync.internal;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Base representation of the frequency a user may choose for sync.
+ *
+ * Configurations should be done using the static initializers for the frequency
+ * type of choice, e.g.:
+ * <p>
+ * <code>
+ * sync.configure(SyncFrequency.newScheduledConfiguration(
+ *    1, TimeUnit.HOURS, true
+ * ));
+ * </code>
+ * </p>
+ */
+public abstract class SyncFrequency {
+  /**
+   * The type of frequency that can be used to schedule sync.
+   */
+  enum SyncFrequencyType {
+    /**
+     * When a change is made locally or a change from a remote collection is received,
+     * the Stitch application will react to the event immediately.
+     * If offline and the change was local, the application will queue the event to
+     * be synchronized when back online.
+     */
+    REACTIVE,
+    /**
+     * Local/remote events will be queued on the device for
+     * a specified amount of time (configurable) before they are applied.
+     */
+    SCHEDULED,
+    /**
+     * The collection will only sync changes when
+     * specified by the application.
+     */
+    ON_DEMAND
+  }
+
+  /**
+   * When a change is made locally or a change from a remote collection is received,
+   * the Stitch application will react to the event immediately.
+   * If offline and the change was local, the application will queue the event to
+   * be synchronized when back online.
+   */
+  static class ReactiveFrequency extends SyncFrequency {
```

Comment:
[q] do these static subclasses need to have the `Frequency` suffix? seems redundant. You would access this class by `SyncFrequency.ReactiveFrequency`, but without the suffix it would be cleaner: `SyncFrequency.Reactive`

Reply:
This is pretty standard Java (or general OOP) naming convention when there is AdjectiveNoun. Exception => FileNotFoundException, Employee => MongoEmployee => StitchEmployee. But I take your point that these are inner subclasses, so I will change them.